app/models/manageiq/providers/microsoft/infra_manager/vm_or_template_shared/scanning.rb

```diff
@@ -1,4 +1,18 @@
 module ManageIQ::Providers::Microsoft::InfraManager::VmOrTemplateShared::Scanning
+  extend ActiveSupport::Concern
+
+  included do
+    supports :smartstate_analysis do
+      if archived?
+        unsupported_reason_add(:smartstate_analysis,
+                               _("Smartstate Analysis cannot be performed on archived #{self.class.model_suffix} VM."))
```

Comment:
Yes I think we can remove it because it is only a text on hover of UI button and we can keep it simple.